memstore/list/vector_party.go

```diff
@@ -38,19 +38,31 @@ type baseVectorParty struct {
 }
 
 // GetOffsetLength returns the <offset, length> pair at ith row.
-func (vp *baseVectorParty) GetOffsetLength(row int) (offset uint32, length uint32) {
+func (vp *baseVectorParty) GetOffsetLength(row int) (offset uint32, length uint32, valid bool) {
 	if vp.offsets == nil || row < 0 || row >= vp.length {
 		return
 	}
 	offset = *(*uint32)(vp.offsets.GetValue(2 * row))
 	length = *(*uint32)(vp.offsets.GetValue(2*row + 1))
+	valid = !(length == 0 && offset == 0)
 	return
 }
 
 // SetOffsetLength update offset/length for nth fow
-func (vp *baseVectorParty) SetOffsetLength(row int, offset, length unsafe.Pointer) {
-	vp.offsets.SetValue(2*row, offset)
-	vp.offsets.SetValue(2*row+1, length)
+func (vp *baseVectorParty) SetOffsetLength(row int, offset, length unsafe.Pointer, valid bool) {
```

Comment:
seems we can get validity from offset and length already, no need for the extra bool argument